### PR TITLE
Add doors

### LINKED ISSUE #20: add Doors
For player to enter (see Mehdi's glass)

### PR DESCRIPTION
**Pull request by npermanne**
Just the addition of doors. They can be open and closed with interact button (E) and destroy by clicking on it.

### CODE REVIEW

File: project.godot
Comment (avivajpeyi) [inline project.godot:1]: is the deletion of phantom-camera, the virtual-joystick accidental?

File: entities/door/door.gd [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+func _on_area_2d_body_entered(body: Node2D) -> void:
+	is_interactible = true
+	print("Press interact(E) to open or click to destroy")
+
+
+func _on_area_2d_body_exited(body: Node2D) -> void:
+	is_interactible = false
+
+
+func _input(event: InputEvent) -> void:
```
Comment (avivajpeyi) [inline entities/door/door.gd]: Do we need another button? Can we just open the door when the player tries to walk into ie (ie collides with it)?

Fewer controls the better...

Plus, adding an extra button mean more headaches for making it work on mobile

File: project.godot
Comment (npermanne) [reply project.godot:1]: Yup I think it is. A strange error message appeared when I first loaded the project which involved these 2

File: entities/door/door.gd (hunk 1 above)
Comment (npermanne) [reply entities/door/door.gd]: I don't know, I can certainly remove it if you prefer.

File: project.godot
Comment (avivajpeyi) [reply project.godot:1]: As these add-ons are being used, I suggest we keep them in. But once this is fixed, merge your stuff in :)

**Pull request merged**